x-pack/plugins/enterprise_search/public/applications/__mocks__/shallow_useeffect.mock.ts

```diff
@@ -4,14 +4,18 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
+let mockUmountHandler: () => void;
```

Comment:
@JasonStoltz Why do we need to add this / modify this file instead of directly importing `unmountHandler` from this file and calling that?

Reply:
Or to maybe ask in a more straightforward manner - what about the previous version of `unmountHandler` was not working for you?

Reply:
Derp sorry, probably should have read your commit message. 🤦‍♀️ That makes sense, although it seems odd for a test to be tracking the # of times useEffect is being called - in a shallow scenario that # is never going to be accurate or true to production anyway since mocking useEffect doesn't correctly capture update restrictions (e.g. only updating when a certain var changes).

Some minor change requests for this:

- `Umount` -> `Unmount`
- Can we combine where we declare the `let` holder and the export, so:

```suggestion
import { useEffect } from 'react';

// Helper for calling returned useEffect unmount functions
 let mockUnmountHandler: () => void; 
export const unmountHandler = () => mockUnmountHandler(); 
```

and remove lines 13-14

Reply:
It matters in a scenario like this:
```
    const handler = (event: StorageEvent) => {
      if (event.key === key) {
        updateFromStorage();
      }
    };
    window.addEventListener('storage', handler);
    return () => window.removeEventListener('storage', handler);
```

If I want to make sure `addEventListener` and `removeEventListener` are called with the exact same handler, it's hard to reason about if there are multiple calls to `addEventListener`, when there really wouldn't be in reality. I struggled to debug that scenario for a while in my test.

Reply:
Hm, in a scenario like that where having useEffect behave realistically instead of simplically matters, why not use `mount`+[unmount](https://enzymejs.github.io/enzyme/docs/api/ReactWrapper/unmount.html instead of `shallow`?

That being said, I do think this is a good & fairly small change, so no objections to keeping it in 👍 I'm just wondering now if we're potentially digging ourselves a hole by trying to extend/rely on useEffect in shallow too much. Although hopefully in a perfect world Enzyme eventually comes up with good baked-in behavior for `shallow`+`useEffect` anyway and we can delete our own custom mock for this. 😄

Reply:
I lost time because of the way this mock was coded. I fixed it so others don't lose time as well. If you have a specific objection to this change, let me know.

Reply:
841675a2ef3ddcb97c3be30747825cb1263496af
6e34e28a43b348e01fb60851907cdbc1ba51e0de
3ad595fddeb127b8673d963fbe7039549a2104a8

Reply:
Hey hey, just wanted to say that I'm sorry if I'm not clearly communicating well in my comments here, and super appreciate you responding to my ever-eternal need to ask "why?" and also ramble at length about things unrelated to my actual change requests 🤦‍♀️ . Changes look great and I have no specific objections, thanks so much for improving this mock!